po-docs/ja/virt-make-fs.1
.\" Automatically generated by Podwrapper::Man 1.40.2 (Pod::Simple 3.35)
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  \*(C+ will
.\" give a nicer C++.  Capital omega is used to do unbreakable dashes and
.\" therefore won't be available.  \*(C` and \*(C' expand to `' in nroff,
.\" nothing in troff, for use with C<>.
.tr \(*W-
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
.    ds C`
.    ds C'
'br\}
.\"
.\" Escape single quotes in literal strings from groff's Unicode transform.
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\"
.\" If the F register is >0, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.SS), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.\"
.\" Avoid warning from groff about undefined register 'F'.
.de IX
..
.nr rF 0
.if \n(.g .if rF .nr rF 1
.if (\n(rF:(\n(.g==0)) \{\
.    if \nF \{\
.        de IX
.        tm Index:\\$1\t\\n%\t"\\$2"
..
.        if !\nF==2 \{\
.            nr % 0
.            nr F 2
.        \}
.    \}
.\}
.rr rF
.\" ========================================================================
.\"
.IX Title "virt-make-fs 1"
.TH virt-make-fs 1 "2019-02-07" "libguestfs-1.40.2" "Virtualization Support"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.if n .ad l
.nh
.SH "名前"
.IX Header "名前"
virt-make-fs \- tar アーカイブまたはファイルからファイルシステムの作成
.SH "書式"
.IX Header "書式"
.Vb 1
\& virt\-make\-fs [\-\-options] input.tar output.img
\&
\& virt\-make\-fs [\-\-options] input.tar.gz output.img
\&
\& virt\-make\-fs [\-\-options] directory output.img
.Ve
.SH "説明"
.IX Header "説明"
Virt-make-fs is a command line tool for creating a filesystem from a tar
archive or some files in a directory.  It is similar to tools like
\&\fBmkisofs\fR\|(1), \fBgenisoimage\fR\|(1) and \fBmksquashfs\fR\|(1).  Unlike those tools,
it can create common filesystem types like ext2/3 or \s-1NTFS,\s0 which can be
useful if you want to attach these filesystems to existing virtual machines
(eg. to import large amounts of read-only data to a \s-1VM\s0).
.PP
To create blank disks, use \fBvirt\-format\fR\|(1).  To create complex layouts,
use \fBguestfish\fR\|(1).
.PP
基本的な使用法は次のとおりです:
.PP
.Vb 1
\& virt\-make\-fs input output.img
.Ve
.PP
where \f(CW\*(C`input\*(C'\fR is either a directory containing files that you want to add,
or a tar archive (either uncompressed tar or gzip-compressed tar); and
\&\fIoutput.img\fR is a disk image.  The input type is detected automatically.
The output disk image defaults to a raw ext2 sparse image unless you specify
extra flags (see \*(L"\s-1OPTIONS\*(R"\s0 below).
.SS "ファイルシステム形式"
.IX Subsection "ファイルシステム形式"
The default filesystem type is \f(CW\*(C`ext2\*(C'\fR.  Just about any filesystem type that
libguestfs supports can be used (but \fInot\fR read-only formats like
\&\s-1ISO9660\s0).  Here are some of the more common choices:
.IP "\fIext3\fR" 4
.IX Item "ext3"
Note that ext3 filesystems contain a journal, typically 1\-32 \s-1MB\s0 in size.  If
you are not going to use the filesystem in a way that requires the journal,
then this is just wasted overhead.
.IP "\fIntfs\fR または \fIvfat\fR" 4
.IX Item "ntfs または vfat"
Useful if exporting data to a Windows guest.
.IP "\fIminix\fR" 4
.IX Item "minix"
Lower overhead than \f(CW\*(C`ext2\*(C'\fR, but certain limitations on filename length and
total filesystem size.
.PP
\fI例\fR
.IX Subsection "例"
.PP
.Vb 1
\& virt\-make\-fs \-\-type=minix input minixfs.img
.Ve
.SS "\s-1TO PARTITION OR NOT TO PARTITION\s0"
.IX Subsection "TO PARTITION OR NOT TO PARTITION"
Optionally virt-make-fs can add a partition table to the output disk.
.PP
Adding a partition can make the disk image more compatible with certain
virtualized operating systems which don't expect to see a filesystem
directly located on a block device (Linux doesn't care and will happily
handle both types).
.PP
On the other hand, if you have a partition table then the output image is no
longer a straight filesystem.  For example you cannot run \fBfsck\fR\|(8)
directly on a partitioned disk image.  (However libguestfs tools such as
\&\fBguestfish\fR\|(1) and \fBvirt\-resize\fR\|(1) can still be used).
.PP
\fI例\fR
.IX Subsection "例"
.PP
\&\s-1MBR\s0 パーティションを追加します:
.PP
.Vb 1
\& virt\-make\-fs \-\-partition \-\- input disk.img
.Ve
.PP
If the output disk image could be terabyte-sized or larger, it's better to
use an EFI/GPT\-compatible partition table:
.PP
.Vb 1
\& virt\-make\-fs \-\-partition=gpt \-\-size=+4T \-\-format=qcow2 input disk.img
.Ve
.SS "\s-1EXTRA SPACE\s0"
.IX Subsection "EXTRA SPACE"
Unlike formats such as tar and squashfs, a filesystem does not \*(L"just fit\*(R"
the files that it contains, but might have extra space.  Depending on how
you are going to use the output, you might think this extra space is wasted
and want to minimize it, or you might want to leave space so that more files
can be added later.  Virt-make-fs defaults to minimizing the extra space,
but you can use the \fI\-\-size\fR flag to leave space in the filesystem if you
want it.
.PP
An alternative way to leave extra space but not make the output image any
bigger is to use an alternative disk image format (instead of the default
\&\*(L"raw\*(R" format).  Using \fI\-\-format=qcow2\fR will use the native qemu/KVM qcow2
image format (check your hypervisor supports this before using it).  This
allows you to choose a large \fI\-\-size\fR but the extra space won't actually be
allocated in the image until you try to store something in it.
.PP
Don’t forget that you can also use local commands including \fBresize2fs\fR\|(8)
and \fBvirt\-resize\fR\|(1) to resize existing filesystems, or rerun virt-make-fs
to build another image from scratch.
.PP
\fI例\fR
.IX Subsection "例"
.PP
.Vb 1
\& virt\-make\-fs \-\-format=qcow2 \-\-size=+200M input output.img
.Ve
.SH "オプション"
.IX Header "オプション"
.IP "\fB\-\-help\fR" 4
.IX Item "--help"
簡単なヘルプを表示します。
.IP "\fB\-\-floppy\fR" 4
.IX Item "--floppy"
仮想フロッピーディスクを作成します。
.Sp
今のところ、容量 (1440K)、パーティション形式 (\s-1MBR\s0) およびファイルシステム形式 (\s-1VFAT\s0)
が事前に選択されています。将来的に、ジオメトリーを選択できるようになる可能性があります。
.IP "\fB\-\-size=\fRN" 4
.IX Item "--size=N"
.PD 0
.IP "\fB\-\-size=+\fRN" 4
.IX Item "--size=+N"
.IP "\fB\-s\fR N" 4
.IX Item "-s N"
.IP "\fB\-s\fR \fB+\fRN" 4
.IX Item "-s +N"
.PD
Use the \fI\-\-size\fR (or \fI\-s\fR) option to choose the size of the output image.
.Sp
If this option is \fInot\fR given, then the output image will be just large
enough to contain all the files, with not much wasted space.
.Sp
To choose a fixed size output disk, specify an absolute number followed by
b/K/M/G/T/P/E to mean bytes, Kilobytes, Megabytes, Gigabytes, Terabytes,
Petabytes or Exabytes.  This must be large enough to contain all the input
files, else you will get an error.
.Sp
To leave extra space, specify \f(CW\*(C`+\*(C'\fR (plus sign) and a number followed by
b/K/M/G/T/P/E to mean bytes, Kilobytes, Megabytes, Gigabytes, Terabytes,
Petabytes or Exabytes.  For example: \fI\-\-size=+200M\fR means enough space for
the input files, and (approximately) an extra 200 \s-1MB\s0 free space.
.Sp
Note that virt-make-fs estimates free space, and therefore will not produce
filesystems containing precisely the free space requested.  (It is much more
expensive and time-consuming to produce a filesystem which has precisely the
desired free space).
.IP "\fB\-\-format=\fR\s-1FMT\s0" 4
.IX Item "--format=FMT"
.PD 0
.IP "\fB\-F\fR \s-1FMT\s0" 4
.IX Item "-F FMT"
.PD
出力するディスクイメージ形式を選択します。
.Sp
デフォルトは \f(CW\*(C`raw\*(C'\fR です (スパースな raw 形式のディスクイメージ)。
.IP "\fB\-\-type=\fR\s-1FS\s0" 4
.IX Item "--type=FS"
.PD 0
.IP "\fB\-t\fR \s-1FS\s0" 4
.IX Item "-t FS"
.PD
出力ファイルシステム形式を選択します。
.Sp
デフォルトは \f(CW\*(C`ext2\*(C'\fR です。
.Sp
Any filesystem which is supported read-write by libguestfs can be used here.
.IP "\fB\-\-label=\fR\s-1LABEL\s0" 4
.IX Item "--label=LABEL"
Set the filesystem label.
.IP "\fB\-\-partition\fR" 4
.IX Item "--partition"
.PD 0
.IP "\fB\-\-partition=\fR\s-1PARTTYPE\s0" 4
.IX Item "--partition=PARTTYPE"
.PD
このフラグが指定されていると、出力ディスクイメージに \s-1MBR\s0 パーティションテーブルを追加します。
.Sp
You can change the partition table type, eg. \fI\-\-partition=gpt\fR for large
disks.
.Sp
For \s-1MBR,\s0 virt-make-fs sets the partition type byte automatically.
.IP "\fB\-v\fR" 4
.IX Item "-v"
.PD 0
.IP "\fB\-\-verbose\fR" 4
.IX Item "--verbose"
.PD
デバッグ情報を有効にします。
.IP "\fB\-V\fR" 4
.IX Item "-V"
.PD 0
.IP "\fB\-\-version\fR" 4
.IX Item "--version"
.PD
バージョン番号を表示して、終了します。
.IP "\fB\-x\fR" 4
.IX Item "-x"
Enable libguestfs trace.
.SH "関連項目"
.IX Header "関連項目"
\&\fBguestfish\fR\|(1), \fBvirt\-format\fR\|(1), \fBvirt\-resize\fR\|(1), \fBvirt\-tar\-in\fR\|(1),
\&\fBmkisofs\fR\|(1), \fBgenisoimage\fR\|(1), \fBmksquashfs\fR\|(1), \fBmke2fs\fR\|(8),
\&\fBresize2fs\fR\|(8), \fBguestfs\fR\|(3), http://libguestfs.org/.
.SH "著者"
.IX Header "著者"
Richard W.M. Jones http://people.redhat.com/~rjones/
.SH "COPYRIGHT"
.IX Header "COPYRIGHT"
Copyright (C) 2010\-2019 Red Hat Inc.
.SH "LICENSE"
.IX Header "LICENSE"
.SH "BUGS"
.IX Header "BUGS"
To get a list of bugs against libguestfs, use this link:
https://bugzilla.redhat.com/buglist.cgi?component=libguestfs&product=Virtualization+Tools
.PP
To report a new bug against libguestfs, use this link:
https://bugzilla.redhat.com/enter_bug.cgi?component=libguestfs&product=Virtualization+Tools
.PP
When reporting a bug, please supply:
.IP "\(bu" 4
The version of libguestfs.
.IP "\(bu" 4
Where you got libguestfs (eg. which Linux distro, compiled from source, etc)
.IP "\(bu" 4
Describe the bug accurately and give a way to reproduce it.
.IP "\(bu" 4
Run \fBlibguestfs\-test\-tool\fR\|(1) and paste the \fBcomplete, unedited\fR
output into the bug report.


ChangeLog
commit 91698a8500d19e06811133bfab27d1e1e65a583e
Author: Richard W.M. Jones <rjones@redhat.com>
Date:   Thu Feb 7 14:45:14 2019 +0000

    Version 1.40.2.

commit c7b374a757821bbd8f369fa268a75485163b52d7
Author: Pino Toscano <ptoscano@redhat.com>
Date:   Wed Feb 6 15:19:53 2019 +0100

    inspect: revamp icon extraction for *SUSE guests
    
    Newer versions do not have the 24px distributor.png icon; OTOH they have
    the 48px version, so look for that one before the 24px one.
    
    Also, bump the size limit to 10K, as newer versions of the icon are
    bigger than 3K.
    
    (cherry picked from commit f41a75cea67dc3f2e18700303fac5d7078b3291a)

commit 8e62f30293737472ec6b9a2a238bb1206b2fbf4e
Author: Pino Toscano <ptoscano@redhat.com>
Date:   Wed Feb 6 15:15:08 2019 +0100

    inspect: factor out find_png helper function
    
    Isolate the code for looking for a PNG file among a specified list from
    the icon_ubuntu implementation, and switch that function to it.
    
    This is just code refactoring, with no behaviour changes.
    
    (cherry picked from commit 6ff95acc72b02c4df0520e8cae7190447e4b2da5)

commit 178d9d821af6ee25f9f5141c20e4de42315de25f
Author: Pino Toscano <ptoscano@redhat.com>
Date:   Wed Feb 6 15:05:51 2019 +0100

    inspect: bump size limit for Mageia guests
    
    Newer versions of the logo in Cauldron are almost 3K, so bump the limit
    to 10K for now.
    
    (cherry picked from commit 032b8d4c72ab233922482871403c2d9fe2f4a75b)

commit cc83aae48dbda40afa1b1652bd1a53dc9dd61573
Author: Pino Toscano <ptoscano@redhat.com>
Date:   Tue Jan 29 15:07:18 2019 +0100

    v2v: linux: use NEVR for querying RPM packages (RHBZ#1669395)
    
    Use NEVR when querying RPM for the list of files of a package, instead
    of ENVR.  Also, use the epoch only when non-zero, and version of RPM
    supports it.
    
    The approach is basically copied from what supermin does in its RPM
    package handler.
    
    (cherry picked from commit 5eafd1273132b79018ddb72426b128f0fb8a39f4)

commit 76797064449a99351ca44586b16a50c5b74c0a47
Author: Richard W.M. Jones <rjones@redhat.com>
Date:   Mon Jan 21 15:59:45 2019 +0000

    build: Reduce -Wformat-overflow=2 to =1.
    
    GCC 9 gives this error:
    
    qemuopts.c: In function 'qemuopts_to_config_channel':
    qemuopts.c:987:29: error: '%.*s' directive output between 0 and 2147483647 bytes may exceed minimum required size of 4095 [-Werror=format-overflow=]
      987 |             fprintf (fp, "  %.*s = ", (int) k, values[j]);
          |                             ^~~~
    qemuopts.c:987:26: note: assuming directive output of 1 byte
      987 |             fprintf (fp, "  %.*s = ", (int) k, values[j]);
          |                          ^~~~~~~~~~~
    
    See discussion at the following GCC bug:
    
    https://gcc.gnu.org/bugzilla/show_bug.cgi?id=88993
    (cherry picked from commit b26bd778b58b78eb59503413fd5b41ac23725690)

commit d5dc8fba74edb39bede2e42316f3bd0e9c26c24c
Author: Pino Toscano <ptoscano@redhat.com>
Date:   Tue Jan 22 13:04:59 2019 +0100

    python: change types for RBufferOut/FBuffer with Python 3 (RHBZ#1661871)
    
    So far RBufferOut return values, and FBuffer struct fields are 'str' on
    all the versions of Python.  Python 3 distinguishes between 'str'
    (unicode strings), and 'bytes', with 'str' no more able to hold
    arbitrary data.
    
    For this reason, switch the return value of RBufferOut functions, and
    FBuffer struct fields to bytes on Python 3: while this is a potentially
    incompatibile change, this is the only way to handle safely sequences
    of arbitrary bytes.
    
    (cherry picked from commit 0ee02e0117527b86a31b2a88a14994ce7f15571f)

commit c18957e0ea4233e2237e68c7d5e647cbde69e764
Author: Pino Toscano <ptoscano@redhat.com>
Date:   Tue Jan 22 12:43:11 2019 +0100

    python: fix call of Python handlers of events
    
    Make sure to reference the arguments, to make sure they are kept alive
    during the function call; this is visible when setting an event handler
    for the CLOSE event, and testing it with Python 3.
    
    This does not seem to create a memory leak e.g. with Python 2.
    
    Also, switch away from the quasi-internal PyEval_CallObject to the
    public PyObject_CallObject, which takes care of doing safety checks.
    
    (cherry picked from commit 85235aec837716f1ddb2926b9a59a02543195500)

commit 53a7f2ffec1f7b014db4909371e0c1b1576ba10a
Author: Pino Toscano <ptoscano@redhat.com>
Date:   Fri Jan 18 13:22:26 2019 +0100

    build: integrate ocaml-link.sh with automake silent rules
    
    In case either the silent rules are disabled, or V=1 is set, print the
    full command line that is executed.
    
    (cherry picked from commit a8006afaf545d5188f592d8583f9dc373ded7671)

commit 1f026244ba5891a00617f4ab8c41b8680ff8c497
Author: Richard W.M. Jones <rjones@redhat.com>
Date:   Thu Jan 17 12:06:36 2019 +0000

    Version 1.40.1.

commit ae2aa14527605ffb89d86bf93bf8ca4f969118e1
Author: Richard W.M. Jones <rjones@redhat.com>
Date:   Thu Jan 17 12:03:07 2019 +0000

    valgrind: Add workaround for slow loading of debuginfo by valgrind.
    
    See nbdkit commit
    https://github.com/libguestfs/nbdkit/commit/bd32d899edb4c592a2df0ce4b26a2cf8396eba1d
    and https://bugzilla.redhat.com/show_bug.cgi?id=1662656
    
    (cherry picked from commit ea7c13fed7922b233db8382b1348dfb436e3ba85)

commit 1de72f8f00600bcaa76742dcfd366fb702eef3c2
Author: Richard W.M. Jones <rjones@redhat.com>
Date:   Tue Jan 15 14:39:41 2019 +0000

    Version 1.40.0.

commit 6151d8694d9db0d709a9ad807a5ca6a098d6502c
Author: Richard W.M. Jones <rjones@redhat.com>
Date:   Wed Jan 16 11:49:45 2019 +0000

    po: Drop ca and pt_BR from linguas.
    
    Commit 4252a490bb117940c1d38c838345ed110457386e dropped the PO files
    for ca (Catalan) and pt_BR (Brazilian Portuguese).  However it did not
    drop these languages from the linguas line in the Makefile resulting

po-docs/uk/virt-make-fs.pl
=encoding utf8

=head1 НАЗВА

virt-make-fs - Make a filesystem from a tar archive or files

=head1 КОРОТКИЙ ОПИС

 virt-make-fs [--options] input.tar output.img

 virt-make-fs [--options] input.tar.gz output.img

 virt-make-fs [--options] directory output.img

=head1 ОПИС

Virt-make-fs is a command line tool for creating a filesystem from a tar
archive or some files in a directory.  It is similar to tools like
L<mkisofs(1)>, L<genisoimage(1)> and L<mksquashfs(1)>.  Unlike those tools,
it can create common filesystem types like ext2/3 or NTFS, which can be
useful if you want to attach these filesystems to existing virtual machines
(eg. to import large amounts of read-only data to a VM).

To create blank disks, use L<virt-format(1)>.  To create complex layouts,
use L<guestfish(1)>.

Basic usage is:

 virt-make-fs input output.img

where C<input> is either a directory containing files that you want to add,
or a tar archive (either uncompressed tar or gzip-compressed tar); and
C<output.img> is a disk image.  The input type is detected automatically.
The output disk image defaults to a raw ext2 sparse image unless you specify
extra flags (see L</OPTIONS> below).

=head2 ТИП ФАЙЛОВОЇ СИСТЕМИ

The default filesystem type is C<ext2>.  Just about any filesystem type that
libguestfs supports can be used (but I<not> read-only formats like
ISO9660).  Here are some of the more common choices:

=over 4

=item I<ext3>

Note that ext3 filesystems contain a journal, typically 1-32 MB in size.  If
you are not going to use the filesystem in a way that requires the journal,
then this is just wasted overhead.

=item I<ntfs> або I<vfat>

Useful if exporting data to a Windows guest.

=item I<minix>

Lower overhead than C<ext2>, but certain limitations on filename length and
total filesystem size.

=back

=head3 ПРИКЛАД

 virt-make-fs --type=minix input minixfs.img

=head2 TO PARTITION OR NOT TO PARTITION

Optionally virt-make-fs can add a partition table to the output disk.

Adding a partition can make the disk image more compatible with certain
virtualized operating systems which don't expect to see a filesystem
directly located on a block device (Linux doesn't care and will happily
handle both types).

On the other hand, if you have a partition table then the output image is no
longer a straight filesystem.  For example you cannot run L<fsck(8)>
directly on a partitioned disk image.  (However libguestfs tools such as
L<guestfish(1)> and L<virt-resize(1)> can still be used).

=head3 ПРИКЛАД

Додавання розділу MBR:

 virt-make-fs --partition -- input образ.img

If the output disk image could be terabyte-sized or larger, it's better to
use an EFI/GPT-compatible partition table:

 virt-make-fs --partition=gpt --size=+4T --format=qcow2 input disk.img

=head2 EXTRA SPACE

Unlike formats such as tar and squashfs, a filesystem does not "just fit"
the files that it contains, but might have extra space.  Depending on how
you are going to use the output, you might think this extra space is wasted
and want to minimize it, or you might want to leave space so that more files
can be added later.  Virt-make-fs defaults to minimizing the extra space,
but you can use the I<--size> flag to leave space in the filesystem if you
want it.

An alternative way to leave extra space but not make the output image any
bigger is to use an alternative disk image format (instead of the default
"raw" format).  Using I<--format=qcow2> will use the native QEmu/KVM qcow2
image format (check your hypervisor supports this before using it).  This
allows you to choose a large I<--size> but the extra space won't actually be
allocated in the image until you try to store something in it.

Don't forget that you can also use local commands including L<resize2fs(8)>
and L<virt-resize(1)> to resize existing filesystems, or rerun virt-make-fs
to build another image from scratch.

=head3 ПРИКЛАД

 virt-make-fs --format=qcow2 --size=+200M input output.img

=head1 ПАРАМЕТРИ

=over 4

=item B<--help>

Показати коротку довідку.

=item B<--version>

Показати дані щодо версії і завершити роботу.

=item B<--debug>

Увімкнути виведення діагностичних даних.

=item B<--floppy>

Create a virtual floppy disk.

Currently this preselects the size (1440K), partition type (MBR) and
filesystem type (VFAT).  In future it may also choose the geometry.

=item B<--size=E<lt>NE<gt>>

=item B<--size=+E<lt>NE<gt>>

=item B<-s E<lt>NE<gt>>

=item B<-s +E<lt>NE<gt>>

Use the I<--size> (or I<-s>) option to choose the size of the output image.

If this option is I<not> given, then the output image will be just large
enough to contain all the files, with not much wasted space.

To choose a fixed size output disk, specify an absolute number followed by
b/K/M/G/T/P/E to mean bytes, Kilobytes, Megabytes, Gigabytes, Terabytes,
Petabytes or Exabytes.  This must be large enough to contain all the input
files, else you will get an error.

To leave extra space, specify C<+> (plus sign) and a number followed by
b/K/M/G/T/P/E to mean bytes, Kilobytes, Megabytes, Gigabytes, Terabytes,
Petabytes or Exabytes.  For example: I<--size=+200M> means enough space for
the input files, and (approximately) an extra 200 MB free space.

Note that virt-make-fs estimates free space, and therefore will not produce
filesystems containing precisely the free space requested.  (It is much more
expensive and time-consuming to produce a filesystem which has precisely the
desired free space).

=item B<--format=E<lt>fmtE<gt>>

=item B<-F E<lt>fmtE<gt>>

Choose the output disk image format.

The default is C<raw> (raw sparse disk image).

For other choices, see the L<qemu-img(1)> manpage.  The only other choice
that would really make sense here is C<qcow2>.

=item B<--type=E<lt>fsE<gt>>

=item B<-t E<lt>fsE<gt>>

Choose the output filesystem type.

Типовим значенням є C<ext2>.

Any filesystem which is supported read-write by libguestfs can be used here.

=item B<--label=E<lt>LABELE<gt>>

Set the filesystem label.

=item B<--partition>

=item B<--partition=E<lt>parttypeE<gt>>

If specified, this flag adds an MBR partition table to the output disk
image.

You can change the partition table type, eg. I<--partition=gpt> for large
disks.

Note that if you just use a lonesome I<--partition>, the Perl option parser
might consider the next parameter to be the partition type.  For example:

 virt-make-fs --partition input.tar output.img

would cause virt-make-fs to think you wanted to use a partition type of
C<input.tar> which is completely wrong.  To avoid this, use I<--> (a double
dash) between options and the input and output arguments:

 virt-make-fs --partition -- input.tar output.img

For MBR, virt-make-fs sets the partition type byte automatically.

=back

=head1 ЕКРАНУВАННЯ СИМВОЛІВ У ОБОЛОНЦІ

Libvirt guest names can contain arbitrary characters, some of which have
meaning to the shell such as C<#> and space.  You may need to quote or
escape these characters on the command line.  See the shell manual page
L<sh(1)> for details.

=head1 ТАКОЖ ПЕРЕГЛЯНЬТЕ

L<guestfish(1)>, L<virt-format(1)>, L<virt-resize(1)>, L<virt-tar-in(1)>,
L<mkisofs(1)>, L<genisoimage(1)>, L<mksquashfs(1)>, L<mke2fs(8)>,
L<resize2fs(8)>, L<guestfs(3)>, L<Sys::Guestfs(3)>,
L<http://libguestfs.org/>.

=head1 АВТОР

Richard W.M. Jones L<http://people.redhat.com/~rjones/>

=head1 АВТОРСЬКІ ПРАВА

© Red Hat Inc., 2010–2012

